vault/resource_ssh_secret_backend_role.go

```diff
@@ -100,6 +100,11 @@ func sshSecretBackendRoleResource() *schema.Resource {
 			Type:     schema.TypeString,
 			Optional: true,
 		},
+		"default_user_template": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  false,
```

Comment:
I think we can avoid setting a default here and let Vault handle that if it needs a default. If Vault returns a default value, then it would be set to the TF state on the read method

This way would generally help us avoid setting a value to the user's TF state if it isn't required, and we wouldn't need to update the other tests with checks